Develop finished Kaltura Player

vadimkononov opened the pull request.

vadimkononov committed:
kaltura player translate from android

vadimkononov committed:
Update .gitignore

vadimkononov committed:
moving KalturaPlaybackRequestAdapter to KalturaPlayer
deleting 'useStaticMediaProvider'

vadimkononov committed:
expose kaltura player api

vadimkononov committed:
implement example

vadimkononov committed:
default UI manager and controls (play, pause, seek)

vadimkononov committed:
kaltura phoenix player

vadimkononov committed:
ovp player without session provider
backend error event in ott player

vadimkononov committed:
change target of the file

vadimkononov committed:
pass nil to kava from ott

vadimkononov committed:
changes due moving AnonymousSession objects

vadimkononov committed:
use kaltura player with ks=nil

vadimkononov committed:
UIConfManager

vadimkononov committed:
rename UIConf to PlayerConfig

vadimkononov committed:
moving pluginConfig to playerOptions

vadimkononov committed:
change partnerId type to Int

vadimkononov committed:
save uiconf on disk

vadimkononov committed:
implement expiration logic on player config object

vadimkononov committed:
put header in files

vadimkononov committed:
remove dependency on OVP from OTT

vadimkononov committed:
netkit pod updated

vadimkononov committed:
remove Pods folder

vadimkononov committed:
Update .gitignore

vadimkononov committed:
Merge branch 'master' into develop

vadimkononov committed:
Merge branch 'master' into develop

vadimkononov committed:
kaltura stats plugin added as default in kaltura player

vadimkononov committed:
merge plugin configs

vadimkononov committed:
using PKPluginMerge protocol

vadimkononov committed:
use ads in ott example

vadimkononov committed:
implement token replacer

vadimkononov committed:
add token replacer in call

vadimkononov committed:
update pod

vadimkononov committed:
do not cache uiConf in PlayerConfigManager

vadimkononov committed:
ask delegate for prepare() if 'preload' didn't occur

vadimkononov committed:
tracks selection

vadimkononov committed:
move file

vadimkononov committed:
remove chromecast option from ui

vadimkononov committed:
formats added as possible parameter

vadimkononov committed:
trackSelectionSettings added

vadimkononov committed:
version added

vadimkononov committed:
accessibility identifier added to uislider

vadimkononov committed:
update uiconf parse

vadimkononov committed:
uiconf parse

vadimkononov committed:
update kaltura player due to changes in kava

x-NR-x committed:
FEC-9910 Kaltura Player - Basic Flow (#3)

* FEM-2604 KalturaPlayer remodeling. Starting Basic Flow.

* FEM-2604 WIP - Basic Flow

* FEC-9910 Basic Flow.

* Fixed Kava plugin not initialized.
Made public some properties.

* Removed objc.

* Added iOSTestApp, updated travis with the travis-build.

* Updated the Readme.

* Removed media data from the BasicPlayerOptions class.
Added a PKMediaEntry property to be set.
Added a setupMediaEntry function in the BasicPlayer to set the mediaEntry from parameters.
Removed registration to the Kava plugin in the BasicPlayer, only an impression is sent.
Removed the prepare from the KalturaBasicPlayer class into the KalturaPlayer class.
Changed the setPlayerView function to a kalturaPlayerView property.
Added preload, autoplay and startTime cases.
Renamed player property to pkPlayer for better understanding.

* FEC-9959 Added an option to send the mediaType in the setupMediaEntry function.
Set the startTime to be optional, and creating the MediaConfig with the startTime only if it was set.
Removed the play after prepare in case of autoPlay and added an observer to the PlayerEvent.CanPlay and set it there, otherwise the currentItem is not set and in case of a live media it won't seek to the live edge.

* Removed calling play only after canPlay event, this was not the issue and will interfere with play immediately functionality.

* FEC-9910 Added an option to update the PlayerOptions for the player.

* Removed the old example, the samples can be found in the samples repo.

* Updated and added missing documentation.

* FEC-9910 Removed the 'from' naming of the id param.

* Renamed kalturaPlayerView property to view.
Removed check on the didSet of the mediaEntry for same object.

* FEC-9910 Added MediaOptions.
Moved startTime from the player options to the media options, updated relevant code accordingly.
Set the mediaEntry in the KalturaPlayer as internal and added a setMedia function in the BasicKalturaPlayer which can be set with the media options as well.
Updated the KalturaBasicPlayer initializer naming for a cleaner reading approach.

kaltura-hooks commented:
Hi @vadimkononov,
Thank you for contributing this pull request!
Please sign the Kaltura CLA so we can review and merge your contribution.
Learn more at http://bit.ly/KalturaContrib

x-NR-x committed:
FEC-9909 KalturaOTTPlayer (#4)

* FEC-9909 KalturaOTTPlayer WIP

* FEC-9909 OTTKalturaPlayer WIP.
KalturaPlayer will support from iOS 10.
Added dependency to PlayKitProviders and PlayKitKava.
Added OTTDMSConfigModel to save the DMS configuration.
Update the playerOptions pluginConfig upon the call performed to updatePluginConfig.
Added KalturaPlayerError for the player errors.
Added logic for fetching the DMS configuration from the server or the cached data.
Embedded Kava.

* FEC-9909 KalturaOTTPlayer WIP.
Added embedded PhoenixAnalytics.
Updated some functions too internal.

* FEC-9909 Updated and added missing documentation.
Added retry for the DMS Configuration.

* FEC-9909 Removed 'DMS' naming from the core classes and updated code accordingly.

* FEC-9909 Upon creating the player, in case the config data has not yet returned, setting the kava with a default value.

* FEC-9909 Removed BasicPlayerOptions and OTTPlayerOptions, merged to PlayerOptions.
Updated code accordingly.

* FEC-9909 Moved the setup static function to the KalturaOTTPlayer class from the KalturaOTTPlayerManager class.
Updated code accordingly.

* FEC-9909 Added a KavaHelper class to return the plugin config, because it is common for both OTT and OVP.
Updated code accordingly.
Added the Common folder to the OTT subspec.

noamtamim committed:
Update KalturaPlayerManager.swift

x-NR-x committed:
FEC-9872 IMA & IMA DAI Plugin (#5)

* FEC-9872 Added KnownPlugins class.
Register all known PlayKit plugins if the pod was set.
Updated code accordingly.
Code rearranging in KalturaPlayerManager.

* FEC-9872 Added KalturaBasicPlayerManager, this is for the basic player in order for the known plugins to be registered automatically.
Updated documentation.
Fixed IMA class names.

* FEC-9872 Moved the setup function to the KalturaBasicPlayer from the KalturaBasicPlayerManager.
Removed the checking for debug mode, crash if the model doesn't exist.
Fixed the naming of the typealias events.

* FEC-9872 Removed code that was added twice from the merge.

* FEC-9872 Removed check for DEBUG mode, if the function doesn't exist in the sub class it should crash.
